src/gt4py/next/program_processors/runners/dace_fieldview/gtir_builtin_translators.py

```diff
@@ -327,7 +332,7 @@ def translate_broadcast_scalar(
     assert cpm.is_ref_to(stencil_expr, "deref")
 
     domain = extract_domain(domain_expr)
-    field_dims, field_shape = _get_field_shape(domain)
+    field_dims, field_offset, field_shape = _get_field_layout(domain)
```

Comment:
In the end these are symbols, so who does provide them?
You always take the domain's lower bound however, are you sure that this will always work?
It could technically be that for some temporary more must be computed.
This is the case for example in `apply_diffusion_to_vn()` and others there we have the following pattern:
```python
tmp = foo(a)
for i in range(horizontal_start, horizontal_end):
    output[i] = tmp[E2C[i, 0]] + tmp[E2C[i, 1]]
```
In this case `tmp` must fully materialize on the whole domain, because we (currently) does not know what is actually accessed.
So the correct offset here is zero.

Reply:
Yes, this is supported. This GTIR will materialize in a temporary for `tmp` defined as a field operator on the whole domain.

Reply:
This was not my point.
This was an example to showcase, that if you took always the same domain, for example the output domain, then you will have problems.
However, I now realized that you probably set the offset for each transient differently, which is even more strange in my opinion.
Because now a specific value of an index has different meanings depending on which array we put it in.

Reply:
Correct, each transient has its own array offset. The test case I added gives an exhaustive example, with one transient defined in the positive range and one defined in the negative range, both with shape `size - 4`:
```
let_lambda(x, y, size) {
  y @ c⟨ IDimₕ: [2, size - 2) ⟩
       ← (λ(xᐞ1) → 
            (λ(xᐞ2) → 
               (λ(xᐞ3) → 
                  (λ(xᐞ4) → 
                     (λ(xᐞ5) → as_fieldop(λ(__arg0, __arg1) → ·__arg0 + ·__arg1, c⟨ IDimₕ: [2, size - 2) ⟩)(xᐞ5, x))(
                       as_fieldop(λ(__arg0, __arg1) → ·__arg0 + ·__arg1, c⟨ IDimₕ: [2, size - 2) ⟩)(xᐞ3, xᐞ4)
                     ))(as_fieldop(λ(a) → ·⟪IDimₒ, 1000000000ₒ⟫(a), c⟨ IDimₕ: [2, size - 2) ⟩)(xᐞ2)))(
                 as_fieldop(λ(a) → ·⟪IDimₒ, -1000000000ₒ⟫(a), c⟨ IDimₕ: [2, size - 2) ⟩)(xᐞ1)
               ))(
              as_fieldop(
                λ(__arg0, __arg1) → ·__arg0 × ·__arg1, c⟨ IDimₕ: [2 + 1000000000, size + 2 + 1000000000) ⟩
              )(3.0,
                as_fieldop(λ(a) → ·⟪IDimₒ, -1000000000ₒ⟫(a), c⟨ IDimₕ: [2 + 1000000000, size + 2 + 1000000000) ⟩)(x))
            ))(as_fieldop(
                 λ(__arg0, __arg1) → ·__arg0 × ·__arg1, c⟨ IDimₕ: [2 - 1000000000, size - 2 - 1000000000) ⟩
               )(4.0,
                 as_fieldop(λ(a) → ·⟪IDimₒ, 1000000000ₒ⟫(a), c⟨ IDimₕ: [2 - 1000000000, size - 2 - 1000000000) ⟩)(x)));
}
```